src/server/api/services/EventProcessor.ts

```diff
@@ -228,14 +231,41 @@ export class ObservationProcessor implements EventProcessor {
 
   async checkForAlerts(projectId: string, observationId: string) {
     try {
-      const totalTokenCost = await getTokenCostForUser(prisma, {
+      // Check for COST_PER_USER alerts
+      const tokenCost = await getTokenCostForUser(prisma, {
         projectId: projectId,
         observationId: observationId,
       });
-      console.log(
-        `Total token cost for observation ${observationId} is ${totalTokenCost}`,
-      );
-      // TODO: Query cost alerts and check if we triggered any
+      // TODO: This will fire the alert redundantly once the threshold is crossed.
+      //      We should only fire it once per user per time frame.
+      const alertsToTrigger = await prisma.alert.findMany({
+        where: {
+          projectId: projectId,
+          alertMetric: AlertMetric.COST_PER_USER,
+          alertThreshold: {
+            lte: tokenCost.totalTokenCost,
+          },
+        },
+      });
+      alertsToTrigger.forEach((alert) => {
+        if (!alert.triggerWebhookUrl) return;
+        console.log(
+          `[Alerts] Triggering webhook ${alert.triggerWebhookUrl} for alert`,
+          alert.id,
+        );
+        void fetch(alert.triggerWebhookUrl, {
```

Comment:
We deploy our server on vercel, which executes our backend on cloud functions. As we are not waiting for the fetch to complete, i am wondering whether the function could close before the fetch was succeeded resulting in `499`. Do you know whether we can run into this issue?